actor/src/main/scala/org/apache/pekko/actor/LightArrayRevolverScheduler.scala

```diff
@@ -193,6 +195,12 @@ class LightArrayRevolverScheduler(config: Config, log: LoggingAdapter, threadFac
       task
     }
 
+  private def checkPeriod(delay: FiniteDuration): Unit =
```

Comment:
I think @He-Pin made a comment here before which now seems to be gone regarding inlining.

I just wanted to say you can inline this method if you want, you just have to make a `scala-2` version using the `@inline` annotation and a `scala-3` version with the `inline` keyword.

The easiest way to do this would be to place the `checkPeriod` function into a `private[pekko]` trait in the `scala-2`/`scala-3` source folders respectively and make `class LightArrayRevolverScheduler` extend that trait.

Reply:
Yes, we can do that in a separate pr, I have to say, this is a hot path.

Reply:
Hotspot Compiler is very good at optimising this. There is zero evidence that this code needs excess optimisation.

Reply:
> Hotspot Compiler is very good at optimising this. There is zero evidence that this code needs excess optimisation.

I am also skeptical but it can be proved with benchmarks in a separate PR if need be